main.go

```diff
@@ -446,6 +447,26 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager) webhooks.ClusterCac
 		os.Exit(1)
 	}
 
+	// Setup a separate cache for the metadata watches to secrets.
+	// This way the watch does not use the LabelSelector defined at the cache which
+	// would filter to secrets having the cluster label, because secrets referred
+	// by ClusterResourceSet are not specific to a single cluster.
+	partialSecretCache, err := cache.New(mgr.GetConfig(), cache.Options{
+		Scheme:            mgr.GetScheme(),
+		Mapper:            mgr.GetRESTMapper(),
+		HTTPClient:        mgr.GetHTTPClient(),
+		SyncPeriod:        ptr.To(time.Minute * 10),
+		DefaultNamespaces: watchNamespaces,
+	})
+	if err != nil {
+		setupLog.Error(err, "failed to create cache for metadata only Secret watches")
```

Comment:
Sorry now it's back to upper case (because these are log statements not errors now) (same below)